### PR TITLE
feat: Add variable create_route to control creation of route

### PR DESCRIPTION
**Pull request by IlyesDemineExtVeolia**
## Description

Fixes https://github.com/terraform-aws-modules/terraform-aws-apigateway-v2/issues/97

## Motivation and Context

Can be useful to have a variable to control creation of route

## Breaking Changes

No breaking changes

## How Has This Been Tested?
- [ ] I have updated at least one of the `examples/*` to demonstrate and validate my change(s)
- [ ] I have tested and validated these changes using one or more of the provided `examples/*` projects

- [ ] I have executed `pre-commit run -a` on my pull request

### CODE REVIEW

**Comment by IlyesDemineExtVeolia** [comment]
Hello Need review please 🙂 @antonbabenko

File: main.tf [hunk 1]
```diff
@@ -90,7 +90,7 @@ resource "aws_apigatewayv2_stage" "default" {
   }
 
   dynamic "route_settings" {
-    for_each = { for k, v in var.integrations : k => v if var.create_routes_and_integrations && length(setintersection(["data_trace_enabled", "detailed_metrics_enabled", "logging_level", "throttling_burst_limit", "throttling_rate_limit"], keys(v))) > 0 }
+    for_each = { for k, v in var.integrations : k => v if var.create_routes_and_integrations && (!contains(keys(v), "create_route") || v.create_route == true) && length(setintersection(["data_trace_enabled", "detailed_metrics_enabled", "logging_level", "throttling_burst_limit", "throttling_rate_limit"], keys(v))) > 0 }
```
Comment (antonbabenko) [inline main.tf:93]:
```suggestion
    for_each = { for k, v in var.integrations : k => v if var.create_routes_and_integrations && try(tobool(v.create_route), true) && length(setintersection(["data_trace_enabled", "detailed_metrics_enabled", "logging_level", "throttling_burst_limit", "throttling_rate_limit"], keys(v))) > 0 }
```

**Review by antonbabenko — APPROVED** [review]
LGTM, there is just a minor improvement.

**Comment by IlyesDemineExtVeolia** [comment]
Thanks for your review

**Comment by antonbabenko** [comment]
This PR is included in version 3.1.0 :tada:

**Pull request merged**